File: sdks/pkgcloud/challenges/authenticate_token.js
var pkgcloud = require('pkgcloud');

// create our client with your rackspace credentials
var client = pkgcloud.providers.rackspace.compute.createClient({
  username: process.env.RAX_USERNAME,
  apiKey: process.env.RAX_API_KEY,
  authUrl: process.env.OS_AUTH_URL,
  region: process.env.RAX_REGION
});
client.auth(function(err) {
  if (err) {
    console.log(err.message);
    process.exit(1);
  } else {
    console.log('Authenticated');
  }
});

File: sdks/openstack.net/challenges/AuthenticateToken.cs
using System;
using net.openstack.Providers.Rackspace;
using net.openstack.Core.Providers;
using net.openstack.Core.Exceptions.Response;
using net.openstack.Providers.Rackspace.Objects;


namespace openstack.net
{
	class AuthenticateToken : Challenge
	{
		public int Run (string[] args)
		{
			var auth_url = new Uri(Environment.GetEnvironmentVariable ("OS_AUTH_URL"));
			Console.WriteLine ("Connecting to " + auth_url);
			IIdentityProvider identityProvider = new CloudIdentityProvider (null, null, null, auth_url);

			// In this example we're getting our credentials from environment variables
			// asdf

			identityProvider.Authenticate (new RackspaceCloudIdentity {
				Username = Environment.GetEnvironmentVariable("RAX_USERNAME"),
				APIKey = Environment.GetEnvironmentVariable("RAX_API_KEY")
			});
			Console.WriteLine ("Authenticated!");
			return 0;
		}
	}
}
